app/forms/prior_authority/decision_form.rb

```diff
@@ -0,0 +1,76 @@
+module PriorAuthority
+  class DecisionForm
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+    include ActiveRecord::AttributeAssignment
+
+    attribute :submission
+    attribute :current_user
+    attribute :pending_decision, :string
+    attribute :pending_granted_explanation, :string
+    attribute :pending_part_grant_explanation, :string
+    attribute :pending_rejected_explanation, :string
+
+    STATES = [
+      GRANTED = 'granted'.freeze,
+      PART_GRANT = 'part_grant'.freeze,
+      REJECTED = 'rejected'.freeze
+    ].freeze
+
+    validates :pending_decision, inclusion: { in: STATES }
+    validates :pending_rejected_explanation, presence: true, if: -> { pending_decision == REJECTED }
+    validate :check_adjustments_made_if_part_grant
+    validate :not_yet_assessed
+
+    def summary
+      @summary ||= BaseViewModel.build(:application_summary, submission)
+    end
+
+    def save
+      return false unless valid?
+
+      PriorAuthorityApplication.transaction do
+        stash
+        previous_state = submission.state
+
+        submission.update!(state: pending_decision)
+        Event::Decision.build(submission: submission,
+                              comment: explanation,
+                              previous_state: previous_state,
+                              current_user: current_user)
+        NotifyAppStore.process(submission:)
+      end
+
+      true
+    end
+
+    def stash
```

Comment:
do we really need this ability to stash a decision?

Reply:
I couldn't think of a better way to implement `Save and come back later`.  I didn't want to start adding a 'draft' state to Event objects, and I didn't want to put these values in the session of the user's device as the use case was a caseworker drafting a decision and then a supervisor signing off on it. So the blob seemed the best place for it.